class.jetpack-sync-settings.php

```diff
@@ -0,0 +1,623 @@
+	 *
+	 * @return string unmapped site url
+	 */
+	static function get_site_url(){
+
+		return get_site_url( get_current_blog_id() );
+	}
+
+	/**
+	 * Get Sharing Service options
+	 * @param  string $key of the value to get back.
+	 * @return $value
+	 */
+	static function sharing_service( $key ) {
+
+		// This should be part of the whole thing.
+		if ( class_exists( 'Sharing_Service' ) ) {
+			$ss = new Sharing_Service();
+			$sharing = $ss->get_global_options();
+
+			if ( isset( $sharing[ $key ] ) ) {
+				return $sharing[ $key ];
+			}
+		}
+
+		return null;
+	}
+
+	/**
+	 * Get the whether videopress is enabled.
+	 *
+	 * @return bool
+	 */
+	static function videopress_enabled() {
+
+		if ( get_option( 'video_upgrade' ) == '1' ) {
+			return true;
+		} else {
+			if ( class_exists( 'Jetpack_Options' ) ) {
+				$videopress = Jetpack_Options::get_option( 'videopress', array() );
+
+				if ( isset( $videopress[ 'blog_id' ] ) && $videopress[ 'blog_id' ] > 0 ) {
+					return true;
+				}
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Get Options lets you change the name of the get option to be something else.
+	 *
+	 * @param  string $key
+	 * @param  string $default
+	 * @return option value
+	 */
+	static function get_option( $key, $default = false ) {
+		return get_option( $key, $default );
+	}
+
+	/**
+	 * Get a list of suported  the post formats.
+	 * @return array A list of post formats
+	 */
+	static function post_formats() {
+		// deprecated - see separate endpoint. get a list of supported post formats
+		$all_formats       = get_post_format_strings();
+		$supported         = get_theme_support( 'post-formats' );
+
+		$supported_formats = array();
+
+		if ( isset( $supported[0] ) ) {
+			foreach ( $supported[0] as $format ) {
+				$supported_formats[ $format ] = $all_formats[ $format ];
+			}
+		}
+
+		return $supported_formats;
+	}
+
+	/**
+	 * List of allowed mime types
+	 * @return [type] [description]
+	 */
+	static function get_mime_types() {
+		if ( function_exists( 'get_mime_types' ) ) {
+			$allowed_file_types = get_mime_types();
+		} else {
+			// http://codex.wordpress.org/Uploading_Files
+			$mime_types = get_allowed_mime_types();
+			foreach ( $mime_types as $type => $mime_type ) {
+				$extras = explode( '|', $type );
+				foreach ( $extras as $extra ) {
+					$allowed_file_types[] = $extra;
+				}
+			}
+		}
+		return $allowed_file_types;
+	}
+
+	/**
+	 * Tells is if the domain is mapped.
+	 *
+	 * @return boolean [description]
+	 */
+	static function is_mapped_domain(){
+		if ( function_exists( 'get_primary_redirect' ) ) {
+			$primary_redirect = strtolower( get_primary_redirect() );
+			if ( false === strpos( $primary_redirect, '.wordpress.com' ) ) {
+				return true;
+			}
+		}
+		return false;
+	}
+	/**
+	 * Get whether likes is enabled for the site.
+	 *
+	 * @return bool
+	 */
+	static function default_likes_enabled() {
+		return apply_filters( 'wpl_is_enabled_sitewide', ! get_option( 'disabled_likes' ) );
```

Comment:
You'd need to add something like this before the filter:

`/** This filter is documented in modules/likes.php */`